src/main/java/uk/gov/ons/ctp/common/event/EventPublisherImpl.java

```diff
@@ -0,0 +1,86 @@
+package uk.gov.ons.ctp.common.event;
+
+import com.godaddy.logging.Logger;
+import com.godaddy.logging.LoggerFactory;
+import java.util.Date;
+import java.util.UUID;
+import lombok.Getter;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import uk.gov.ons.ctp.common.error.CTPException;
+import uk.gov.ons.ctp.common.event.model.Header;
+import uk.gov.ons.ctp.common.event.model.RespondentAuthenticatedEvent;
+import uk.gov.ons.ctp.common.event.model.RespondentAuthenticatedResponse;
+import uk.gov.ons.ctp.common.event.model.SurveyLaunchedEvent;
+import uk.gov.ons.ctp.common.event.model.SurveyLaunchedResponse;
+
+/** Service responsible for the publication of events. */
+public class EventPublisherImpl implements EventPublisher {
+
+  private static final Logger log = LoggerFactory.getLogger(EventPublisherImpl.class);
+
+  private RabbitTemplate template;
+
+  @Getter
+  public enum EventType {
```

Comment:
Can't the enum be private?

Reply:
Very handy for testing to use for expected results. Means tests will not be as brittle using static strings, you test the code sets as per expected configuration of the enum.